app/Console/Commands/ParseTextFileCommand.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use App\Report;
+use App\Document;
+use Illuminate\Support\Facades\Storage;
+
+class ParseTextFileCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'parse:file {filePath}';
+    protected $filePath;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Parses a text file, and outputs some stuff';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        if (!$this->setPath()) {
+            return;
+        };
+
+        $this->line('Opening file at: ' . $this->filePath);
+
+        $reportPath = $this->generateReport();
+        $this->line('Complete!');
+        $this->line('report was created at: ' . $reportPath);
+    }
+
+    public function setPath(){
+        //Should probably find out how to reference the app root directory, re '/../../../'
+        $this->filePath = realpath(__DIR__ . '/../../../' . $this->argument('filePath'));
```

Comment:
You could just use `app_path()` to get the app root. Make sure you check Laravel helpers first. (https://laravel.com/docs/5.7/helpers -- see paths for things like `public_path()`, `storage_path()`, `resource_path()`, `base_path()`, `config_path()`...etc.) If these helper methods weren't available, you should still consider abstracting this logic out of this file. (env variable or similar) Your code should be readable.